.github/workflows/deploy_railway_ensnode_service.yml

```
@@ -0,0 +1,109 @@
+                echo "SVC_ID="${{ secrets.SUBGRAPH_SVC_ID }} >> "$GITHUB_ENV"
+                ;;
+              *)
+                echo "Branch not recognized, skipping workflow"
+                exit 1
+            esac
+
+      - name: Build & Push Docker image for ENSNode to Github Container registry
+        env:
+          IMAGE_NAME: ghcr.io/${{ github.repository }}/ensnode
+        run: |
+          echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+          docker build -f apps/ensnode/Dockerfile -t $IMAGE_NAME:$TAG .
+          docker push $IMAGE_NAME:$TAG
+
+      - name: Determine Railway variable to be updated
+        run: |
+          TAG=${{ env.TAG }}
+          VARIABLE_NAME=${TAG^^}_DATABASE_SCHEMA
+          echo "Looking for: ${VARIABLE_NAME} environment variable"
+          echo "VARIABLE_NAME=$VARIABLE_NAME" >> "$GITHUB_ENV"
+      
+      - name: Get current Railway schema
+        run: |
+          # Send Railway API request to get all shared variables. We need those to check if we really need to redeploy.
+          API_RESPONSE=`curl --request POST \
+          --fail \
+          --silent \
+          --url https://backboard.railway.app/graphql/v2 \
+          --header 'Project-Access-Token: '${{ env.RAILWAY_TOKEN }} \
+          --header 'Content-Type: application/json' \
+          --data '{
+            "query": "query variables($projectId: String!, $environmentId: String!) { variables(projectId: $projectId, environmentId: $environmentId) }",
+              "variables": {
+                "projectId": "'${{ env.RAILWAY_PROJECT_ID }}'" ,
+                "environmentId": "'${{ env.RAILWAY_ENVIRONMENT_ID }}'"
+              }
+          }'`
+          # Filter all env variables from response except one that is defining schema for specific deployment.
+          RAILWAY_SCHEMA=`echo ${API_RESPONSE} | jq '.data.variables.'${{ env.VARIABLE_NAME }}`
+          echo "RAILWAY_SCHEMA=$RAILWAY_SCHEMA" >> "$GITHUB_ENV"
+          
+
+      #TODO FOR NOW JUST REPLACE github.sha. EXTEND LOGIC IN FUTURE
+      - name: Update Railway schema variable
+        # Check if newly calculated schema matches old one, if not replace env variable and force deployment. Otherwise skip deploymen.
+        run: |
+          # Sending Railway API request to update specifc shared variable
+          if [[ ${{env.RAILWAY_SCHEMA}} != ${{ github.sha }} ]]; then
+          curl --request POST \
+              --silent \
+              --fail \
+              --url https://backboard.railway.app/graphql/v2 \
+              --header 'Project-Access-Token: '${{ env.RAILWAY_TOKEN }} \
+              --header 'Content-Type: application/json' \
+              --data '{"query": "mutation variableUpsert { variableUpsert(input: { projectId: \"'${{ env.RAILWAY_PROJECT_ID }}'\", environmentId: \"'${{ env.RAILWAY_ENVIRONMENT_ID }}'\", name: \"'${VARIABLE_NAME}'\", value: \"${{ github.sha }}\" }) }"}'
```

Perhaps we should define database schema value on the job `env` so it's easier to notice it and also easier to apply any updates to it (i.e. adding env var hash):
```yaml
jobs:
  image-build-and-push:
    runs-on: ubuntu-latest
    env:
      RAILWAY_TOKEN: ${{ secrets.RAILWAY_TOKEN }}
      # Token for redeploy. Seems lika a bug in Railway ..
      RAILWAY_TEAM_TOKEN: ${{ secrets.RAILWAY_TEAM_TOKEN }}
      RAILWAY_PROJECT_ID: ${{ secrets.RAILWAY_PROJECT_ID }}
      RAILWAY_ENVIRONMENT_ID: ${{ secrets.RAILWAY_ENVIRONMENT_ID}}
      DATABASE_SCHEMA: ${{ github.sha }} # this one could also have some env vars hash attached
```

```suggestion
              --data '{"query": "mutation variableUpsert { variableUpsert(input: { projectId: \"'${{ env.RAILWAY_PROJECT_ID }}'\", environmentId: \"'${{ env.RAILWAY_ENVIRONMENT_ID }}'\", name: \"'${VARIABLE_NAME}'\", value: \"${{ env.DATABASE_SCHEMA }}\" }) }"}'
```

---

Nice suggestion. Applied